### PR TITLE
Use ClusterIssuer in SC and disable in WC

### LINKED ISSUE #287: [2] Switch to ClusterIssuers
**What should be investigated.**

Are new service cluster looks as follows:

```
$ kubectl get issuers,clusterissuers -o wide -A
NAMESPACE             NAME                                              READY   STATUS                                                 AGE
elastic-system        issuer.cert-manager.io/letsencrypt-prod           True    The ACME account was registered with the ACME server   4m2s
elastic-system        issuer.cert-manager.io/letsencrypt-staging        True    The ACME account was registered with the ACME server   4m2s
elastic-system        issuer.cert-manager.io/opendistro-es-ca           True    Signing CA verified                                    4m
elastic-system        issuer.cert-manager.io/opendistro-es-selfsigned   True                                                           4m
influxdb-prometheus   issuer.cert-manager.io/letsencrypt-prod           True    The ACME account was registered with the ACME server   4m2s
influxdb-prometheus   issuer.cert-manager.io/letsencrypt-staging        True    The ACME account was registered with the ACME server   4m2s
kube-system           issuer.cert-manager.io/letsencrypt-staging        True    The ACME account was registered with the ACME server   4m2s
monitoring            issuer.cert-manager.io/letsencrypt-staging        True    The ACME account was registered with the ACME server   4m2s
```

This feels fragile and wasteful. Can we use two ClusterIssuers instead, `letsencrypt-prod` and `letsencrypt-staging`?

**What artifacts should this produce.**

* A working demo with ClusterIssuers or a reason why this is a bad idea.

### PR DESCRIPTION
**Pull request by lentzi90**
**What this PR does / why we need it**:
See #287 for why we want to use ClusterIssuers. I also disabled the (Cluster)Issuers in WC since we don't really need them there. We only used them with kubeapi-metrics for health checking the API server. But this check [doesn't need a trusted certificate](https://github.com/elastisys/compliantkubernetes-apps/blob/main/helmfile/values/blackbox.yaml.gotmpl#L23).

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #287

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:
A nicer solution would be to make the blackbox exporter use `tls_verify` depending on what issuer we use and add a config option to specify [extra annotations for kubeapi-metrics](https://github.com/elastisys/compliantkubernetes-apps/blob/main/helmfile/charts/kubeapi-metrics/values.yaml#L14) so that we can [use an Issuer](https://github.com/elastisys/compliantkubernetes-apps/compare/lj/use-cluster-issuer?expand=1) instead of a ClusterIssuer in WC. This is a bit larger change though and I wasn't sure if it was worth it (especially considering the config overhaul and that blackbox exporter doesn't verify TLS). Should I go for it anyway?

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [x] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.

**Pipeline config** :heavy_check_mark:

### CODE REVIEW

File: config/config/wc-config.yaml [hunk 1]
```diff
@@ -382,17 +382,13 @@ issuers:
...
-    prod:
-      ## Mail through which letsencrypt can contact you.
-      email: "set-me"
-    staging:
-      ## Mail through which letsencrypt can contact you.
-      email: "set-me"
+    enabled: false
+    # prod:
+    #   ## Mail through which letsencrypt can contact you.
+    #   email: null
```
Comment (OlleLarsson) [inline config/config/wc-config.yaml:388]: Just tested and this did not work as expected. Using `""` instead gives a warning.

Comment (OlleLarsson) [reply config/config/wc-config.yaml:388]: With 
```ShellSession
issuers:
  letsencrypt:
    prod:
      email: ""
    staging:
      email: null
```

yq returns nothing/null in bash with `email: ""`, whereas it return the string `null` when `email: null` :facepalm:
Leaving the value empty for a field also works `email:` as yq will return empty string.

Comment (lentzi90) [reply config/config/wc-config.yaml:388]: Oh good to know. I'll change the comment to have `""` instead of `null` then.

Comment (OlleLarsson) [reply config/config/wc-config.yaml:388]: This bothers me as all of these should be equivalent as according to the [YAML specification](https://yaml.org/spec/1.2/spec.html#id2805071)
```ShellSession
Regular expression         Resolved to tag
null | Null | NULL | ~     tag:yaml.org,2002:null
/* Empty */                tag:yaml.org,2002:null
```
```
parent:
  key1:               # empty so "null", # is a comment!
  key2: ~             # also "null"
  key3: null          # "null" explicitly ))
  key4: !!null "null" # for the funs of "secondary tag handle: !!"
  key5: "null"        # sorry, it is a string or !!str if you like ((
```

**Comment by OlleLarsson** [comment]
This looks ok to me.
I'm just a bit worried that we need to use a clusterissuer for user-alertmanager if that's enabled, but i guess if we explain it to the user that they shouldn't use it for their certificates then i think we'll be fine.

Concerning your question about the blackbox api server healthcheck i'd say that it is alright to skip right now and maybe do it in a separate pr.

**Comment by lentzi90** [comment]
Thanks for the comments!
I created an issue for [Issuer/ClusterIssuer](https://github.com/elastisys/compliantkubernetes-apps/issues/311) and mentioned in [this issue](https://github.com/elastisys/compliantkubernetes-apps/issues/207) for the blackbox exporter that we should also verify TLS based on some variable instead of always skipping it.

**Review by OlleLarsson — APPROVED** [review]

**Review by viktor-f — APPROVED** [review]

**Pull request merged**